documentation/OpenMP-4.5-grammar.txt

```diff
@@ -1,114 +1,396 @@
-2.15.3.2 shared -> SHARED (variable-name-list)
-2.15.3.3 private -> PRIVATE (variable-name-list)
-2.15.3.4 firstprivate -> FIRSTPRIVATE (variable-name-list)
-2.15.3.5 lastprivate -> LASTPRIVATE (variable-name-list)
-2.15.3.6 reduction -> REDUCTION(reduction-identifier: variable-name-list)
-2.15.3.7 linear -> LINEAR(linear-list [:linear-step])
-           linear-list -> list | modifier(list)
-           modifier -> REF | VAL | UVAL
-2.15.4.1 copyin -> COPYIN (variable-name-list)
-2.15.4.2 copyprivate -> COPYPRIVATE (variable-name-list)
-2.15.5.1 map -> MAP ([[ALWAYS[,]] (TO | FROM | TOFROM | ALLOC | RELEASE |
+
+2.7.1 collapse-clause -> COLLAPSE (scalar-constant)
+
+2.7.1 ordered-clause -> ORDERED [(scalar-constant)]
+
+2.7.1 nowait-clause -> NOWAIT
+
+2.8.1 aligned-clause -> ALIGNED (variable-name-list[ : scalar-constant])
+
+2.8.1 safelen-clause -> SAFELEN (scalar-constant)
+
+2.8.1 simdlen-clause -> SIMDLEN (scalar-contant)
+
+2.8.2 uniform-clause -> UNIFORM (dummy-arg-name-list)
+
+2.8.2 inbranch-clause -> INBRANCH
+
+2.8.2 notinbranch-clause -> NOTINBRANCH
+
+2.13.9 depend-clause -> DEPEND (((IN | OUT | INOUT) : variable-name-list) |
+                                SOURCE |
+                                SINK : vec)
+                 vec -> iterator [+/- scalar-int-expr],..., iterator[...]
+
+2.9.2 num-tasks-clause -> NUM_TASKS (scalar-int-expr)
+
+2.9.2 grainsize-clause -> GRAINSIZE (scalar-int-expr)
+
+2.9.2 nogroup-clause -> NOGROUP
+
+2.9.2 untied-clause -> UNTIED
+
+2.9.2 priority-clause -> PRIORITY (scalar-int-expr)
+
+2.9.2 mergeable-clause -> MERGEABLE
+
+2.9.2 final-clause -> FINAL (scalar-int-expr)
+
+2.10.1 use-device-ptr-clause -> USE_DEVICE_PTR (variable-name-list)
+
+2.10.1 device-clause -> DEVICE (scalar-integer-expr)
+
+2.10.4 is-device-ptr-clause -> IS_DEVICE_PTR (variable-name-list)
+
+2.10.5 to-clause -> TO (variable-name-list)
+
+2.10.5 from-clause -> FROM (variable-name-list)
+
+2.10.6 link-clause -> LINK (variable-name-list)
+
+2.10.7 num-teams-clause -> NUM_TEAMS (scalar-integer-expr)
+
+2.10.7 thread-limit-clause -> THREAD_LIMIT (scalar-integer-expr)
+
+2.10.8 dist-schedule-clause -> DIST_SCHEDULE (STATIC [ , chunk_size])
+
+2.12 if-clause -> IF ([ directive-name-modifier :] scalar-logical-expr)
+
+2.15.3.1 default-clause -> DEFAULT (PRIVATE | FIRSTPRIVATE | SHARED | NONE)
+
+2.15.3.2 shared-clause -> SHARED (variable-name-list)
+
+2.15.3.3 private-clause -> PRIVATE (variable-name-list)
+
+2.15.3.4 firstprivate-clause -> FIRSTPRIVATE (variable-name-list)
+
+2.15.3.5 lastprivate-clause -> LASTPRIVATE (variable-name-list)
+
+2.15.3.6 reduction-clause -> REDUCTION (reduction-identifier: variable-name-list)
+         reduction-identifier -> + | - | * | .AND. | .OR. | .EQV. | .NEQV. |
+                                 MAX | MIN | IAND | IOR | IEOR
+
+2.15.3.7 linear-clause -> LINEAR (linear-list[ : linear-step])
+         linear-list -> list | modifier(list)
```

Comment:
This production includes the non-terminal "list".  But I don't see "list" defined anywhere.

Reply:
This `list` is like `variable-name-list`.